src/Kernel-Tests-Extended/ValueWithinFixTest.class.st

```diff
@@ -26,6 +26,22 @@ ValueWithinFixTest >> testValueWithinNonLocalReturnFixReal [
 ]
 
 { #category : 'tests' }
+ValueWithinFixTest >> testValueWithinNonLocalReturnFixRealMilliseconds [
+	"The real test for the fix is just as obscure as the original problem"
+
+	| startTime |
+	self valueWithinNonLocalReturn.
+	startTime := [] microsecondClockValue.
+	[[] repeat] valueWithinMilliseconds: 100 onTimeout:[ | deltaTime |
+		"This *should* timeout after 100 msecs but the pending process from
+		the previous invokation will signal timeout after 20 msecs already
+		which will in turn cut this invokation short."
+		deltaTime := [] microsecondClockValue - startTime.
```

Comment:
Super strange to ask the clock to the closure.

Reply:
Indeed. Now this is a test of closure and if we want to run these tests without the time package then I do not see how we can do it.

Reply:
Maybe value with timeout does not need to be kernel? How is it used? I'll check

Reply:
Yes this too. Now having a version with milliseconds was easy and I like to have it this way people can choose what they want and duration is optional.